Functional.Primitives.Extensions/ResultExtensions.cs

```diff
@@ -36,6 +48,38 @@ public static class ResultExtensions
 		public static async Task<Result<TResult, TFailure>> Select<TSuccess, TFailure, TResult>(this Task<Result<TSuccess, TFailure>> result, Func<TSuccess, TResult> select)
 			=> (await result).Select(select);
 
+		public static Result<Option<TResult>, TFailure> SelectIfSome<TSuccess, TFailure, TResult>(this Result<Option<TSuccess>, TFailure> result, Func<TSuccess, TResult> select)
+			=> result.Select(success => success.Match(x => Option.Some(select(x)), Option.None<TResult>));
```

Comment:
Is there difference between this and `result.Select(success => success.Select(select))`?

Reply:
Nope! Good eye. Will make that change.

Reply:
Cool, just had me wondering if that might help with the naming, given that it's multi-type `Select` composition. 

`SelectSelect` is awkward, but `SelectThroughOption` or `SelectOntoOption` might work. It just seems like the `IfSome` is already captured through the meaning of `Select`.

`SelectOption`?

Reply:
Made the change to use `success.Select(select)`

My impression is that `SelectIfSome` reads better, but I have no data to back that up...  I read `SelectIfSome` as "The `select` function that I supply will be executed if the result is successful (because the extension method should follow the rules of `Select`) _and_ the contained `Option<T>` in the success value is `Some`".  `SelectOption` may be confusing given the `Func<TSuccess, TResult>` parameter, but - again - I have no data to back that up.